src/QueryClause.ts

```diff
@@ -0,0 +1,334 @@
+/**
+ * Base QueryClause implementation.
+ */
+export abstract class QueryClause {
+    abstract toJson(): any;
```

Comment:
@yongpingchen Thank you for checking it.

I also checked it and I found [this](https://github.com/Microsoft/TypeScript/issues/5228).  internal modifier is proposed but not implemented.

> Let us keep the current design, we can change it later if there is better solution.

I agree with you.